services/src/main/java/org/keycloak/services/clientpolicy/executor/SecureSigningAlgorithmForSignedJwtEnforceExecutor.java

```diff
@@ -84,7 +83,12 @@ public void executeOnEvent(ClientPolicyContext context) throws ClientPolicyExcep
             case TOKEN_INTROSPECT:
             case LOGOUT_REQUEST:
                 boolean isRequireClientAssertion = Optional.ofNullable(configuration.isRequireClientAssertion()).orElse(Boolean.FALSE).booleanValue();
-                if (!isRequireClientAssertion) break;
+                String authenticatorType = session.getContext().getClient().getClientAuthenticatorType();
+                if (!isRequireClientAssertion && !JWTClientAuthenticator.PROVIDER_ID.equals(authenticatorType)) {
```

Comment:
One small nitpick: Is it possible to check if `client_assertion` parameter is present instead of enforcing `!JWTClientAuthenticator` type? Something like: 
```
if (!isRequireClientAssertion && ObjectUtil.isBlank(clientAssertion))
```

In theory, if someone wants to create his own subclass of JWTClientAuthenticator (with slightly modified behaviour for some better security etc), he won't be able to use this executor if we directly enforce type of authenticator.

Reply:
@mposolda Yes, that's right. I will fix codes as you proposed.

Reply:
@tnorimat Thanks!